공장에서는 시제품의 농도가 60%라고 주장하며 품질관리팀에서 10개의 샘플을 뽑았다. 유의수준 5%에서 다음을 검정하시오.

(1) 귀무가설, 대립가설을 작성하시오

In [1]:
import pandas as pd
import numpy as np
data=pd.DataFrame({"농도":[52, 50, 62, 75, 26, 45, 62, 35, 57, 14]})
data

농도
0  52
1  50
2  62
3  75
4  26
5  45
6  62
7  35
8  57
9  14

H0 : 시제품의 농도는 60%

H1 : 시제품의 농도는 60%가 아니다.

샘플이 10이하이므로 윌콕슨 순위합 검정을 진행한다.

(2) 유효한 샘플의 수를 계산하시오.

: 중위수와 동일한 값들은 순위부호 검정에서 불필요한 데이터이다.

In [8]:
data['농도'].median()

51.0

In [10]:
len(data[data['농도']==51])
#0개임
data_filter=data[data['농도']!=51]

In [11]:
mean0=60
stats.wilcoxon(data_filter['농도']-mean0, alternative='two-sided')

WilcoxonResult(statistic=9.5, pvalue=0.064453125)

검정통계량은 9.5이다. p-value가 0.05보다 크기 떄문에 귀무가설을 기각할 수 없다. 즉 시제품의 농도는 60%라고 할 수 있다.